dotnet/src/Extensions/TemplateEngine.PromptTemplateEngine/Blocks/NamedArgBlock.cs

```diff
@@ -0,0 +1,156 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Microsoft.Extensions.Logging;
+using Microsoft.SemanticKernel.Diagnostics;
+using Microsoft.SemanticKernel.Orchestration;
+
+namespace Microsoft.SemanticKernel.TemplateEngine.Prompt.Blocks;
+
+internal sealed class NamedArgBlock : Block, ITextRendering
+{
+    internal override BlockTypes Type => BlockTypes.NamedArg;
+
+    internal string Name { get; } = string.Empty;
+
+    public string GetValue(ContextVariables? variables)
+    {
+        var valueIsValidValBlock = this._valBlock != null && this._valBlock.IsValid(out var errorMessage);
+        if (valueIsValidValBlock)
+        {
+            return this._valBlock!.Render(variables);
+        }
+
+        var valueIsValidVarBlock = this._argValueAsVarBlock != null && this._argValueAsVarBlock.IsValid(out var errorMessage2);
+        if (valueIsValidVarBlock)
+        {
+            return this._argValueAsVarBlock!.Render(variables);
+        }
+
+        return string.Empty;
+    }
+
+    public NamedArgBlock(string? text, ILoggerFactory? logger = null)
+        : base(NamedArgBlock.TrimWhitespace(text), logger)
+    {
+        var argParts = this.Content.Split(Symbols.NamedArgBlockSeparator);
+        if (argParts.Length > 2)
+        {
+            this.Logger.LogError("Invalid named argument `{0}`.", this.Content);
+            throw new SKException($"Invalid named argument `{this.Content}`. A named argument can contain at most one equals sign separating the arg name from the arg value.");
+        }
+
+        if (argParts.Length == 2)
+        {
+            var trimmedArgParts = NamedArgBlock.GetTrimmedParts(text);
+            this.Name = trimmedArgParts[0];
+            this._argNameAsVarBlock = new VarBlock($"{Symbols.VarPrefix}{trimmedArgParts[0]}");
+            var secondPart = trimmedArgParts[1];
```

Comment:
nit: Consider giving a meaningful name to the expression on the right side of the equal sign. Instead of naming it `secondPart`, use a proper name that reflects its value, such as `argumentValue`.